flang/lib/Optimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -600,6 +590,41 @@ AliasAnalysis::Source AliasAnalysis::getSource(mlir::Value v,
             defOp = v.getDefiningOp();
             return;
```

Comment:
Any idea why the OpenMP case should not follow the same logic and continue the walk?

Reply:
This is why I added @DominikAdamski to the review. To show that we want to set a better precedent for loads and see if maybe  something similar can be done for Omp. I did not quite follow the need for it.